### PR TITLE
Remove spaCy 3.0 deprecation warnings

### PR DESCRIPTION
**Pull request by koaning**
This PR fixes https://github.com/RasaHQ/rasa/issues/8882.

### CODE REVIEW

**Comment by koaning** [comment]
I discovered some `core` tests that passed before and needed to be changed. Figured I should also add a test to make sure that we raise an appropriate error when the user forgets to add a `model` to their `SpacyNLP` config.

**Comment by koaning** [comment]
@jupyterjazz I think you got auto-assigned to this PR. On my end: it's no longer a work in progress so feel free to review. Also, feel free to pass it along to somebody else if you feel the auto-tagging system got it wrong.

File: rasa/nlu/utils/spacy_utils.py [hunk 1]
```diff
@@ -85,62 +82,19 @@ def cache_key(
...
-        )
+        if not component_meta.get("model"):
+            raise InvalidModelError(
+                f"Missing model configuration for `SpacyNLP` in `config.yml`.\n"
+                f"You must pass a model to the `SpacyNLP` component explicitly.\n"
+                f"For example:\n"
+                f"- name: SpacyNLP\n"
+                f"  model: en_core_web_md\n"
+                f"More informaton can be found on {DOCS_URL_COMPONENTS}#spacynlp"
+            )
```
Comment (jupyterjazz) [inline rasa/nlu/utils/spacy_utils.py:94]: Seems good!
Just one question though: shouldn't we also write a test for this error message?

File: tests/nlu/test_spacy_utils.py [hunk 2]
```diff
@@ -5,35 +5,13 @@
...
 def test_model_raises_error_not_exist():
     """It should throw a direct error when a model doesn't exist."""
     with pytest.raises(InvalidModelError):
         SpacyNLP.create({"model": "dinosaurhead"}, RasaNLUModelConfig())
+
+
+def test_model_raises_error_no_model_given():
+    """It should throw a direct error when no model is passed."""
+    with pytest.raises(InvalidModelError):
+        SpacyNLP.create({"model": None}, RasaNLUModelConfig())
```
Comment (jupyterjazz) [inline tests/nlu/test_spacy_utils.py:17]: I'm just wondering if it makes more sense to combine these two tests cause they only differ with an argument value for "model".

File: rasa/nlu/utils/spacy_utils.py (hunk 1 above)
Comment (koaning) [reply rasa/nlu/utils/spacy_utils.py:94]: This is handled by the `test_model_raises_error_no_model_given`-test.

File: tests/nlu/test_spacy_utils.py (hunk 2 above)
Comment (koaning) [reply tests/nlu/test_spacy_utils.py:17]: Fair enough. It's two examples that could be one. I usually prefer to go parameterized once I hit `n=3` but I'll make the change.

File: rasa/nlu/utils/spacy_utils.py (hunk 1 above)
Comment (koaning) [reply rasa/nlu/utils/spacy_utils.py:94]: Ah... you mean you want to test the contents of the message? We could go there but that feels like a bit of overkill.

Comment (jupyterjazz) [reply rasa/nlu/utils/spacy_utils.py:94]: I checked and `test_model_raises_error_no_model_given` covers the error in `load_model` and not this one. Am I missing something?
error

Comment (koaning) [reply rasa/nlu/utils/spacy_utils.py]: Yep. I just checked and fixed it. There was actually a small bug there where it gave the wrong error message. Just fixed it by changing the test and making a fix. 

Good call on your end!

**Comment by koaning** [comment]
@jupyterjazz I've incorporated your requests and they led to a better PR. Does it look good to merge now?

**Review by jupyterjazz — APPROVED** [review]
Thanks! Looks great! 👍

**Pull request merged**